h2o-core/src/main/java/hex/CategoricalEncodingSupport.java

```diff
@@ -0,0 +1,5 @@
+package hex;
+
+public interface CategoricalEncodingSupport {
+    Model.Parameters.CategoricalEncodingScheme[] supportedCategoricalEncodingSchemes();
```

Comment:
1. I'd strongly advise adding Javadoc. Such a crucial interface in our core modules only benefits from having it. Part of the contract described should be whether the returned array is modifiable or not.
2. If Set was returned (ImmutableSet ideally), searching would be faster afterwards. Not sure if this is possible due to our custom serialization, just a general question.